src/SimpleThings/EntityAudit/AuditReader.php

```diff
@@ -819,4 +819,44 @@ public function getEntityHistory($className, $id)
 
         return $result;
     }
+    
+    public function matchRevisions($className, $id, RevisionCriteria $criteria)
```

Comment:
This method isn't matching any revision but returning matched revision in an new array. `matchRevisions` should be renamed to something more meaningful, such as `findRevisionsByCriteria`